### PR TITLE
Add career ladder specification

### PR DESCRIPTION
**Pull request by damianavila**
This is based on the whole conversation in https://github.com/2i2c-org/meta/issues/402, the last few months of experience using some of these pieces (such as the matrix we used for the latest hiring processes), and conversations with team members.
I am purposely trying to reuse existing frameworks and build on top of it. This is why I am referencing some of those pieces via links (I do not think we should copy/paste the content but let me know if you disagree).
Any feedback is welcome!!

### CODE REVIEW

**Comment by sgibson91** [comment]
+1 (Not specifically approving so others can provide feedback too)

**Review by consideRatio — APPROVED** [review]
I think its a great call to adapt existing definitions of axes and levels etc instead of defining that ourselves - it seems very hard to do.

I figure future incremental improvements here could be to:
- define who evaluates step and level increments for the various track members
- define when step and level increments are evaluated

File: engineering/structure.md [hunk 1]
```diff
@@ -22,6 +22,32 @@ They are expected to coordinate with other leads within 2i2c in order to ensure
...
+
+We have an internal and engineering-specific salary bands matrix associated with each level at https://docs.google.com/spreadsheets/d/1D_-RzXhyWw8jFs7KQQXKVRV4VrmOyXjQni2UTnk1044/edit#gid=0.
+
+### EM4
+
+* Evangelizes: researches, creates proofs of concept and introduces new technologies to the team
+* Owns: owns the production operation and monitoring of the system and is aware of its SLAs
+* Coordinates: manages the team members’ career, expectations, performance and level of happiness
+* Challenges: adjusts the team processes, listening to feedback and guiding the team through the changes
+* Team: makes an impact on the whole team, not just on specific parts of it
```
[the remark this answers is not in the document]
Comment (damianavila) [reply engineering/structure.md:49]: Sure, I will add the additional radar plot.

**Comment by damianavila** [comment]
> I think its a great call to adapt existing definitions of axes and levels etc instead of defining that ourselves - it seems very hard to do.

Yep, that is the whole idea underneath this and follow-up PRs.

> I figure future incremental improvements here could be to:

Indeed!! After merging this PR the next step is the mapping of the current team in the proposed structure and then working on the processes for evaluation of levels/steps whereas we define short-term Q4 personal goals (before October) and medium-term 2024 goals (during Q4).

**Comment by damianavila** [comment]
@2i2c-org/engineering, I plan to merge this PR before EOW (tomorrow EOD) if I do not see any blockers. Please comment if you are inclined. Thanks! 

cc @GeorgianaElena and @yuvipanda

**Review by GeorgianaElena — APPROVED** [review]

**Review by yuvipanda — APPROVED** [review]
I am happy we are adapting an existing external ladder for the most part!

File: engineering/structure.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (damianavila) [reply engineering/structure.md]: Added in https://github.com/2i2c-org/team-compass/pull/765/commits/b92e07a9c44e36077742354462ae30acd3822e90

File: engineering/structure.md [hunk 2]
```diff
@@ -22,6 +22,40 @@ They are expected to coordinate with other leads within 2i2c in order to ensure
...
+4. The Tech Lead is a role that can be played by anyone in the team **at different levels**: https://career-ladders.dev/engineering#tech-lead. This is a substantial difference from the dedicated Tead Lead ladder proposed by engineeringladders.com. Our existing [Technology Lead role](./roles/technology-lead.md) represents a Tech Lead with high seniority (D4 and above).
+5. We also introduce **steps** for each numbered level as a way to build salary bands and enhance the granularity. It is expected that engineers advance step by step before entering a new level, although a whole-level jump may exist if there are enough reasons to justify it.
+6. Steps are evaluated every 6 months for 1-3 levels and every year for 4-7 levels, whereas numbered levels are evaluated every year for 1-3 levels and every 2 years for 4-7 levels.
+
+We have an internal and engineering-specific salary bands matrix associated with each level at https://docs.google.com/spreadsheets/d/1D_-RzXhyWw8jFs7KQQXKVRV4VrmOyXjQni2UTnk1044/edit#gid=0.
+
+### EM4
+
+```{image} /images/ladders/em4.png
```
Comment (damianavila) [inline engineering/structure.md:45]:
```suggestion
```{image} /images/em4.png
```

File: engineering/structure.md [hunk 3]
```diff
@@ -22,6 +22,40 @@ They are expected to coordinate with other leads within 2i2c in order to ensure
...
+
+We have an internal and engineering-specific salary bands matrix associated with each level at https://docs.google.com/spreadsheets/d/1D_-RzXhyWw8jFs7KQQXKVRV4VrmOyXjQni2UTnk1044/edit#gid=0.
+
+### EM4
+
+```{image} /images/em4.png
+:class: only-light
+```
+
+```{image} /images/ladders/em4-dark.png
```
Comment (damianavila) [inline engineering/structure.md:49]:
```suggestion
```{image} /images/em4-dark.png
```

**Pull request merged**